scripts/src/main/resources/scripts/command/vscode

```diff
@@ -126,6 +138,18 @@ function doStartVsCode() {
   doRunVsCode "${WORKSPACE_PATH}"
 }
 
+function doAddRecommendations() {
+  doEcho "add recommendations..."
+  local extensions="${WORKSPACE_PATH}/.vscode/extensions.json"
+  if [ -f "${extensions}" ]
+  then
+    rm "${extensions}"
```

Comment:
IMHO this is a dangerous behavior and most users will very much dislike it.
What if some plugins are already installed (active=true, or manually installed) and then (maybe weeks later) a recommended plugin shall be added? So you are simply kicking out all plugins of the user?
That seems odd or maybe I do not understand this approach and am missing something...

Reply:
https://github.com/devonfw/ide/blob/3e4e62f7b1dbf6c079415d279d769c213f43e680/scripts/src/main/resources/scripts/command/vscode#L52-L55

The `doAddRecommendation()` function is only executed during the execution of `vscode setup`. So the `extensions.json` can not be deleted anyway. So we don't need this query there either. We can assume that the `extensions.json` does not exist. 

Furthermore, the `extensions.json` for installed plugins is in `software/vscode-extensions`. Unfortunately, I can't use this file because adding the `recommendations` key there has no effect on vscode. The recommendations have to be added to `${WORKSPACE_PATH}/.vscode/extensions.json`

Reply:
Oh, that sounds like bad news and this IMHO means that this feature is not really working as it is.
Further, implementing it properly will be tricky as new workspaces can be added at any time.

> Unfortunately, I can't use this file because adding the recommendations key there has no effect on vscode.

See
https://github.com/devonfw/ide/blob/c53d140ccc881b869a9e082a599f25a9379e7320/scripts/src/main/resources/scripts/command/vscode#L107
and
https://github.com/devonfw/ide/blob/c53d140ccc881b869a9e082a599f25a9379e7320/scripts/src/main/resources/scripts/command/vscode#L120

I am confused that `extensions.json` are read from `user-data-dir` rather than `extensions-dir`. Bug in VSCode?

Reply:
I have implemented a new solution to add the recommendations. Now after each run of `devon vscode` the plugin recommendation is added to the current workspace. If there are is already an entry with the same values, nothings happens. But when the user adds  new plugins in to the properties file, the `extensions.json` is updated with the new plugins set to `false`.

Reply:
Seems fine now.